index.js

```diff
@@ -311,18 +353,31 @@ class WakeupSession {
 
     peer.unlink(this.peers)
 
-    if (!this.handlers) return
+    for (let i = 0; i < this.sessions.length; i++) {
```

Comment:
might be cleaner if we use `for (const session of this.sessions)` for all of these

Reply:
actually we should traverse backwards for reentry safety, let me fix